src/react/reconcilation.js

```diff
@@ -1070,16 +1000,19 @@ export class Reconciler {
 
           for (let [key, binding] of propsValue.properties) {
             if (binding && binding.descriptor && binding.descriptor.enumerable && key !== "children") {
-              setProperty(newProps, key, getProperty(this.realm, propsValue, key));
+              Properties.Set(this.realm, newProps, key, getProperty(this.realm, propsValue, key), true);
```

Comment:
Why this change? I thought we intentionally used "internal" methods in these cases

Reply:
After speaking to @NTillmann it seems we don't need to when dealing with ObjectValue setting. So I'd rather remove duplicate logic where possible.